pkg/storage/tsdb/block/index.go

```diff
@@ -249,14 +250,15 @@ func GatherIndexHealthStats(logger log.Logger, fn string, minTime, maxTime int64
 
 	// Per series.
 	for p.Next() {
-		lastLset = append(lastLset[:0], lset...)
+		lastLset = lset
```

Comment:
Ok, I see it. No need to make a copy anymore because now we're using the builder which creates a new slice each time we call `builder.Labels()` right?

Reply:
This was mentioned in the commit description: https://github.com/grafana/mimir/pull/3912/commits/14f10ae4ed8ffce1754917b4fac626d2784be3ea